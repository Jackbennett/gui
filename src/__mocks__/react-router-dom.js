import React from 'react';
import { compose, setDisplayName } from 'recompose';
import * as rrd from 'react-router-dom';
// Just render plain div with its children
// eslint-disable-next-line no-import-assign
rrd.BrowserRouter = compose(setDisplayName('BrowserRouter'))(({ children }) => <div>{children}</div>);

module.exports = rrd;
