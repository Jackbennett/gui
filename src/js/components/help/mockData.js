export const helpProps = {
  docsVersion: '',
  findLocalIpAddress: jest.fn,
  isHosted: false,
  isEnterprise: false,
  menderDebPackageVersion: 'master',
  menderVersion: 'master',
  menderArtifactVersion: 'master',
  org: {
    tenant_token: ''
  }
};
