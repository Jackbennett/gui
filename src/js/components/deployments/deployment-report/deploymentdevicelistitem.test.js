import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import DeploymentDeviceListItem from './deploymentdevicelistitem';
import { defaultState, undefineds } from '../../../../../tests/mockData';

describe('DeploymentDeviceListItem Component', () => {
  it('renders correctly', async () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <DeploymentDeviceListItem device={defaultState.deployments.byId.d1.devices.a1} created="2019-01-01" idAttribute="Device ID" viewLog={jest.fn} />
        </MemoryRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
    expect(JSON.stringify(tree)).toEqual(expect.not.stringMatching(undefineds));
  });
});
