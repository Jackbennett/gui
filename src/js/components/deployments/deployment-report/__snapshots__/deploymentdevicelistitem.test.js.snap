// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeploymentDeviceListItem Component renders correctly 1`] = `
<tr
  class="MuiTableRow-root"
>
  <td
    class="MuiTableCell-root"
  >
    <a
      href="/devices?id=a1"
      style="font-weight: 500;"
    >
      <div>
        a1
      </div>
    </a>
  </td>
  <td
    class="MuiTableCell-root"
  >
    -
  </td>
  <td
    class="MuiTableCell-root"
  >
    -
  </td>
  <td
    class="MuiTableCell-root"
  >
    -
  </td>
  <td
    class="MuiTableCell-root"
  >
    -
  </td>
  <td
    class="MuiTableCell-root"
    style="padding-right: 0px; position: relative; min-width: 200px;"
  >
    installing
    <div
      style="position: absolute; bottom: 0px; width: 100%;"
    >
      <div
        class="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-indeterminate"
        role="progressbar"
      >
        <div
          class="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Indeterminate"
        />
        <div
          class="MuiLinearProgress-bar MuiLinearProgress-bar2Indeterminate MuiLinearProgress-barColorPrimary"
        />
      </div>
    </div>
  </td>
  <td
    class="MuiTableCell-root"
  />
</tr>
`;
